src/common/common.cc

```diff
@@ -29,6 +30,23 @@ UniqueID globally_unique_id(void) {
   return result;
 }
 
+/* ObjectID hashing function. */
+size_t hashObjectID(const ObjectID &key) {
+    uint32_t hash = 0;
+    CHECK(UNIQUE_ID_SIZE % sizeof(uint32_t) == 0);
+    for (int i = 0; i < UNIQUE_ID_SIZE / sizeof(uint32_t); i++) {
+      hash ^=  *reinterpret_cast<const uint32_t *>(&key.id[i * sizeof(uint32_t)]);
+    }
+    return std::hash<uint32_t>()(hash);
+}
+/* ObjectID equality function. */
+bool operator==(const ObjectID& x, const ObjectID& y) {
+  if ((*reinterpret_cast<const uint32_t *>(&x.id[0])) !=
+      (*reinterpret_cast<const uint32_t *>(&y.id[0])))
+      return false;
+  return UNIQUE_ID_EQ(x, y);
+}
```

Comment:
Oh, I missed the `if` statement. If we profile this and it's faster then let's keep it this way, otherwise let's just use `UNIQUE_ID_EQ(x, y)`.